testing/src/main/java/io/grpc/testing/StreamRecorder.java

```diff
@@ -127,4 +142,18 @@ public boolean awaitCompletion(int timeout, TimeUnit unit) throws Exception {
   public ListenableFuture<T> firstValue() {
     return firstValue;
   }
+
+  /**
+   * Waits for a value to be received from the stream since the last
+   * call to awaitNext() or construction.  Useful for testing streaming
+   * call patterns.
+   */
+  public void awaitNext() {
+    nextValueLock.lock();
+    if (!seenNextValue) {
```

Comment:
Always use a while loop for conditions. Conditions can have spurious wakeups.

Note that makes it a bit annoying for calling `await(long, TimeUnit)` because the time keeps changing. You can do `nanos = timeUnit.toNanos(time)` and then use `nanos = nextValueCondition.awaitNanos(nanos)`.